_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,270 @@
+---
+layout: post
+title: 'Deploying an OpenID Secured WildFly Application on OpenShift'
+date: 2023-08-04
+tags: oidc authentication OpenShift
+synopsis: How to secure a web application deployed to WildFly with OpenID Connect and deploy it on OpenShift
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+A future release of WildFly will include the ability to add additional scope values when securing applications using
+OpenID Connect. This blog post demonstrates how to secure a WildFly application using OpenID Connect and configure scope
+values to request additional claims using a pre-release version of the feature. We will also walk through how to deploy
+it on the cloud using https://developers.redhat.com/developer-sandbox[Red Hat OpenShift].
+
+toc::[]
+
+== OpenID Connect
+https://openid.net/developers/how-connect-works/[OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. A future release of WildFly will allow the user to configure additional scope values to request specific claim values.
+
+== Prerequisites
+We can deploy this application using Red Hat OpenShift https://developers.redhat.com/developer-sandbox[Developer Sandbox]. There are other https://www.redhat.com/en/technologies/cloud-computing/openshift/try-it[options] available for running OpenShift. Please refer to this https://docs.wildfly.org/29/Getting_Started_on_OpenShift.html[getting started guide] to learn more about WildFly on OpenShift. We will be making use of the following tools for this walkthrough: 
+
+* https://docs.docker.com/[Docker] (you can also use https://docs.podman.io/en/latest/[Podman])  
+* https://developers.redhat.com/developer-sandbox[Developer Sandbox]
+* https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://quay.io/repository/[Quay]
+* https://helm.sh/[Helm]
+
+== Example Application
+For this walkthrough, we will be making use of the https://github.com/PrarthonaPaul/elytron-examples/tree/elytron-oidc-client-scope/elytron-oidc-client-scope[elytron-oidc-client-scope] project which makes use of the ability to configure additional scope values for OpenID Connect. First, clone the `elytron-example` repo locally and navigate to the `elytron-oidc-client-scope` project:
+```
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd elytron-examples/elytron-oidc-client-scope
+```
+
+== Log Into OpenShift Cluster
+
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
```

Comment:
OpenShift CLI should be a link to https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html

Reply:
updated